subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -440,10 +440,33 @@ static int cmd_wifi_disconnect(const struct shell *sh, size_t argc,
 static int cmd_wifi_scan(const struct shell *sh, size_t argc, char *argv[])
 {
 	struct net_if *iface = net_if_get_default();
+	struct wifi_scan_params params = { 0 };
 
 	context.sh = sh;
 
-	if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, NULL, 0)) {
+	if (argc > 2) {
+		shell_fprintf(sh, SHELL_WARNING, "Invalid number of arguments\n");
+		return -ENOEXEC;
+	}
+#ifdef WIFI_MGMT_SCAN_MODE_PASSIVE
+	params.scan_mode = WIFI_SCAN_MODE_PASSIVE;
+#else
+	params.scan_mode = WIFI_SCAN_MODE_ACTIVE;
+#endif
+	if (argc == 2) {
+		if (!strncmp(argv[1], "active", 6)) {
+			params.scan_mode = WIFI_SCAN_MODE_ACTIVE;
+		} else  if (!strncmp(argv[1], "passive", 6)) {
```

Comment:
Default is active, and is using `-p` better, we can use getopt? Its only a flag.